### PR TITLE
Dependency updates

### PR DESCRIPTION
**Pull request by joschi**
* Upgrade to Hibernate Validator 6.0.7.Final
  * https://github.com/hibernate/hibernate-validator/blob/6.0.7.Final/changelog.txt#L4-L56
* Upgrade to Log4j 2.10.0
  * https://logging.apache.org/log4j/2.x/changes-report.html#a2.10.0
* Upgrade to OkHttp 3.9.1
  * https://github.com/square/okhttp/blob/17485bc41852ee9586ebe167555737825a0d1043/CHANGELOG.md#version-391
* Upgrade to Guava 23.6
  * https://github.com/google/guava/releases/tag/v23.5
  * https://github.com/google/guava/releases/tag/v23.6
* Upgrade to EqualsVerifier 2.4
* Upgrade to System Rules 1.17.0
* Upgrade to SpotBugs annotations 3.1.1
  * https://github.com/spotbugs/spotbugs/blob/3.1.1/CHANGELOG.md
* Upgrade to Google Protocol Buffers 3.5.1
  * https://github.com/google/protobuf/releases/tag/v3.5.0
  * https://github.com/google/protobuf/releases/tag/v3.5.1
* Upgrade to Error Prone 2.2.0
  * https://github.com/google/error-prone/releases/tag/v2.2.0
* Upgrade to Mockito 2.13.0
* Upgrade to Google Auto Value 1.5.3 
  * https://github.com/google/auto/releases/tag/auto-value-1.5.3
* Upgrade to AssertJ 3.9.0
  * https://joel-costigliola.github.io/assertj/assertj-core-news.html#assertj-core-3.9.0
* Upgrade to RabbitMQ AMQP Java client 5.1.1
  * https://github.com/rabbitmq/rabbitmq-java-client/releases/tag/v5.1.0
  * https://github.com/rabbitmq/rabbitmq-java-client/releases/tag/v5.1.1
* Upgrade to Dropwizard Metrics 4.0.2
  * https://github.com/dropwizard/metrics/releases/tag/v4.0.0
  * https://github.com/dropwizard/metrics/releases/tag/v4.0.1
  * https://github.com/dropwizard/metrics/releases/tag/v4.0.2
* Upgrade to Elasticsearch 5.6.5  …
  * https://www.elastic.co/guide/en/elasticsearch/reference/5.6/release-notes-5.6.5.html
* Upgrade to SpotBugs Maven Plugin 3.1.1
* Upgrade to Javadoc Maven Plugin 3.0.0

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -98,10 +98,10 @@
         <gelfclient.version>1.4.1</gelfclient.version>
         <grok.version>0.1.7-graylog</grok.version>
         <guava-retrying.version>2.0.0</guava-retrying.version>
-        <guava.version>23.4-jre</guava.version>
+        <guava.version>23.5-jre</guava.version>
         <guice.version>4.1.0</guice.version>
         <HdrHistogram.version>2.1.10</HdrHistogram.version>
-        <hibernate-validator.version>6.0.4.Final</hibernate-validator.version>
+        <hibernate-validator.version>6.0.5.Final</hibernate-validator.version>
```
Comment (bernd) [inline pom.xml]: I am seeing the following log message:

```
2018-01-04 16:38:33,805 INFO : org.hibernate.validator.internal.util.Version - HV000001: Hibernate Validator 5.1.3.Final
```

Is there an issue with transitive dependencies?

Comment (joschi) [reply pom.xml]: This was a result of the artifact relocation into another group:
https://developer.jboss.org/wiki/HibernateValidatorMigrationGuide?_sscc=t#jive_content_id_600Final

File: pom.xml [hunk 2]
```diff
@@ -326,7 +326,7 @@
                     <dependency>
                         <groupId>com.google.errorprone</groupId>
                         <artifactId>error_prone_core</artifactId>
-                        <version>2.1.2</version>
+                        <version>2.1.3</version>
```
Comment (bernd) [inline pom.xml]: @joschi The Jenkins build fails with the following message:

```
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /var/lib/jenkins/workspace/graylog-project-pr-snapshot/.repos/graylog-plugin-cef/src/test/java/org/graylog/plugins/cef/codec/CEFCodecFixturesTest.java:[82,47] [StreamResourceLeak] Streams that encapsulate a closeable resource should be closed using try-with-resources
    (see http://errorprone.info/bugpattern/StreamResourceLeak)
  Did you mean 'final File[] fixtureFiles ;'?
```

So we either have to create a PR for the CEF plugin or disable the rule.

Comment (joschi) [reply pom.xml]: https://github.com/Graylog2/graylog-plugin-cef/pull/25

**Pull request merged**